lib/db.js

```diff
@@ -60,35 +63,76 @@ function normalizeType(params) {
   return type;
 }
 
-class LimitDB {
+class LimitDB extends EventEmitter {
   constructor(params) {
+    super();
     params = params || {};
 
     if (typeof params.path !== 'string' && !params.inMemory) {
       throw new Error('path is required');
     }
-    const path = !params.inMemory ? params.path : undefined;
 
-    this._db = level(path, {
+    const dbPath = !params.inMemory ? params.path : undefined;
+
+    const done = (err, db, types) => {
+      if (err) {
+        return this.emit('error', err);
+      }
+      this._db = db;
+      this._types = types;
```

Comment:
What if someone triggers a "take" or "wait", etc before the ready event? is limitd-server already prepared not to do so? will you send another PR for that?

Reply:
yes, limitd is not binding the TCP server until the database is ready, this prevent blowing up memory.